solr/core/src/test/org/apache/solr/response/TestPushWriter.java

```diff
@@ -45,26 +46,47 @@
 
   public void testStandardResponse() throws IOException {
     ByteArrayOutputStream baos = new ByteArrayOutputStream();
-    OutputStreamWriter osw = new OutputStreamWriter(baos, StandardCharsets.UTF_8);
-    PushWriter pw = new JSONWriter(osw,
-        new LocalSolrQueryRequest(null, new ModifiableSolrParams()), new SolrQueryResponse());
-    writeData(pw);
-    osw.flush();
-    if (log.isInfoEnabled()) {
-      log.info("{}", new String(baos.toByteArray(), StandardCharsets.UTF_8));
+    Map m;
+    try (OutputStreamWriter osw = new OutputStreamWriter(baos, StandardCharsets.UTF_8)) {
+      JSONWriter pw = new JSONWriter(osw,
+          new LocalSolrQueryRequest(null, new ModifiableSolrParams()), new SolrQueryResponse());
+      writeData(null, pw);
+      osw.flush();
+      if (log.isInfoEnabled()) {
+        log.info("{}", new String(baos.toByteArray(), StandardCharsets.UTF_8));
+      }
+      m = (Map) Utils.fromJSON(baos.toByteArray());
+      checkValues(m);
     }
-    @SuppressWarnings({"rawtypes"})
-    Map m = (Map) Utils.fromJSON(baos.toByteArray());
-    checkValues(m);
+
     try (JavaBinCodec jbc = new JavaBinCodec(baos= new ByteArrayOutputStream(), null)) {
       writeData(jbc);
-      try (JavaBinCodec jbcUn = new JavaBinCodec()) {
-        m = (Map) jbcUn.unmarshal(new ByteArrayInputStream(baos.toByteArray()));
-      }
+      m = (Map) Utils.fromJavabin(baos.toByteArray());
     }
     checkValues(m);
   }
 
+  public void testXmlWriter() throws Exception {
+    try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        OutputStreamWriter osw = new OutputStreamWriter(baos, StandardCharsets.UTF_8)) {
+      XMLWriter xml = new XMLWriter(osw,
+          new LocalSolrQueryRequest(null, new ModifiableSolrParams()), new SolrQueryResponse());
```

Comment:
what is this `LocalSolrQueryRequest` for?

Reply:
XmlWriter takes SolrQueryRequest as one of the arg. so, I have created localSolrQueryRequest